src/ripple/rpc/impl/RPCHelpers.cpp

```diff
@@ -696,19 +696,28 @@ parseRippleLibSeed(Json::Value const& value)
 std::optional<Seed>
 getSeedFromRPC(Json::Value const& params, Json::Value& error)
 {
-    // The array should be constexpr, but that makes Visual Studio unhappy.
-    static char const* const seedTypes[]{
-        jss::passphrase.c_str(), jss::seed.c_str(), jss::seed_hex.c_str()};
+    using string_to_seed_t = std::function<std::optional<Seed>(const std::string &)>;
+    using seed_match_t = std::pair<const char *, string_to_seed_t>;
+    
+    static seed_match_t const seedTypes[]{
+        { jss::passphrase.c_str(), [](const std::string &s) { return parseGenericSeed(s); }},
+        { jss::seed.c_str(),       [](const std::string &s) { return parseBase58<Seed>(s); }},
+        { jss::seed_hex.c_str(),   [](const std::string &s) {
+            uint128 i;
+            if (i.parseHex(s))
+                return std::optional<Seed>(Slice(i.data(), i.size()));
+            return std::optional<Seed>{}; }}
+        };
 
     // Identify which seed type is in use.
-    char const* seedType = nullptr;
+    const seed_match_t* seedType = nullptr;
     int count = 0;
-    for (auto t : seedTypes)
+    for (const auto &t : seedTypes)
```

Comment:
I think we generally use east const style, but I may be wrong.

Reply:
Thanks @thejohnfreeman will do (and also set clang-format to run automatically).

Reply:
done!

Reply:
I'm not sure any changes were committed. They don't seem to show up in this PR. If I may request, please don't force push your branches. Just add another commit and prefix the commit message subject with `[fold]` (team practice). That way we can switch between reviewing just the new updates in isolation or all of the changes together. Every `[fold]`-tagged commit will be squashed when the PR is merged.

Reply:
Thanks @thejohnfreeman. Yes the changes are committed. I didn't know about the `[fold]` tag and I will use this from now on. I think if you look at the current diffs you'll see your requested change was made.

Reply:
Ok, I'm not sure what changed then. Here is the change I was requesting:

```cpp
    for (auto const& t : seedTypes)
```

Not `const auto& t` or `const auto &t`, but `auto const& t`. I only flagged this one instance, but there are several more in the changeset. The old code was not entirely consistent itself, but it was mostly in this style known as "[east const](https://hackingcpp.com/cpp/design/east_vs_west_const.html)". Don't need to fix the whole file, but any code that is touched should be fixed. clang-format won't fix or flag any of this because it is not a whitespace change, which is all clang-format does.

Reply:
@thejohnfreeman thanks for explaining. Is the preference you state documented anywhere?

Reply:
east const it is.

Reply:
Yes, it's here: https://github.com/ripple/rippled/blob/develop/docs/CodingStyle.md